src/wzapi.cpp

```diff
@@ -975,30 +975,26 @@ bool wzapi::structureIdle(WZAPI_PARAMS(const STRUCTURE *psStruct))
 	return ::structureIdle(psStruct);
 }
 
-std::vector<const STRUCTURE *> _enumStruct_fromList(WZAPI_PARAMS(optional<int> _player, optional<wzapi::STRUCTURE_TYPE_or_statsName_string> _structureType, optional<int> _looking), STRUCTURE **psStructLists)
+std::vector<const STRUCTURE *> _enumStruct_fromList(WZAPI_PARAMS(optional<int> _player, optional<wzapi::STRUCTURE_TYPE_or_statsName_string> _structureType, optional<int> _playerFilter), STRUCTURE **psStructLists)
 {
 	std::vector<const STRUCTURE *> matches;
-	int player = -1, looking = -1;
 	WzString statsName;
 	STRUCTURE_TYPE type = NUM_DIFF_BUILDINGS;
 
-	player = (_player.has_value()) ? _player.value() : context.player();
+	int player = (_player.has_value()) ? _player.value() : context.player();
```

Comment:
as i'm not the C++ expert. is it valid to remove the reference here to also use value_or?
```cpp
// old
std::string &id = (arg2.has_value()) ? arg2.value() : arg1;
// new
std::string id = arg2.value_or(arg1);
```

Reply:
I think it is better if you leave it as it is in this case. Removing the reference would cause the string to be copied (possibly causing an allocation/deallocation in the heap).